src/app/loadout-drawer/loadout-apply.ts

```diff
@@ -608,6 +645,80 @@ export function clearItemsOffCharacter(
   };
 }
 
+/**
+ * Applies the socket overrides for the passed in loadout items.
+ *
+ * This gets all the sockets for an item and either applies the override plug in the items
+ * socket overrides, or applies the default item plug. If the plug is already in the socket
+ * we don't actually make an API call, it is just counted as a success.
+ */
+function applySocketOverrides(
+  itemsWithOverrides: LoadoutItem[]
+): ThunkResult<{ total: number; successful: number }> {
+  return async (dispatch, getState) => {
+    const defs = d2ManifestSelector(getState())!;
+
+    const overrideResults = { total: 0, successful: 0 };
+
+    for (const item of itemsWithOverrides) {
+      if (item.socketOverrides) {
+        const dimItem = getItemAcrossStores(storesSelector(getState()), { id: item.id })!;
+
+        // We only handle class items at the moment
+        if (dimItem.bucket.type === 'Class') {
+          // First we clear out the fragment sockets as they take up an energy value provided by aspects
+          const fragmentsToClear: { socketIndex: number; mod: PluggableInventoryItemDefinition }[] =
+            [];
+          const fragmentSockets = getSocketsByCategoryHash(
+            dimItem.sockets!,
+            SocketCategoryHashes.Fragments
+          );
+
+          for (const fragmentSocket of fragmentSockets) {
+            const initialPlug = defs.InventoryItem.get(
+              fragmentSocket.socketDefinition.singleInitialItemHash
+            ) as PluggableInventoryItemDefinition;
+            fragmentsToClear.push({ socketIndex: fragmentSocket.socketIndex, mod: initialPlug });
+          }
+
+          // TODO more logging around the clearing of aspects.
+          // If this fails atm I just expect the following code block that assigns overrides to also
+          // fail and notify the user of it
+          await dispatch(equipMods(item.id, fragmentsToClear));
```

Comment:
Note that I am specifically clearing fragments out here so we don't break the energy barrier when changing aspects. I am not sure if this is needed but figured its worth putting in initially so we can get a working version rolled out.